### PR TITLE
feat: [BD-46] implement new pagination component

### PR DESCRIPTION
**Pull request by vlasovmichael**
- Added reduced and minimal variants per the design spec
- Added dark palette option
- Changed exist default variant styles
- Updated Pagination component in DataTable footer

JIRA Pagination: https://openedx.atlassian.net/browse/PAR-371
JIRA Pagination in datatable: https://openedx.atlassian.net/browse/PAR-643

### CODE REVIEW

File: src/DataTable/TablePaginationButtonGroup.jsx [hunk 1]
```diff
@@ -1,35 +1,21 @@
...
-      />
-      <IconButton
-        alt="Next page"
-        src={ArrowForwardIos}
-        iconAs={Icon}
-        onClick={nextPage}
-        disabled={!canNextPage}
-      />
-    </ButtonGroup>
+    <Pagination variant="minimal" currentPage={pageIndex + 1} pageCount={pageCount} paginationLabel="1234" onPageSelect={(pageNum) => gotoPage(pageNum - 1)} />
```
Comment (monteri) [inline src/DataTable/TablePaginationButtonGroup.jsx:18]: If there are more than 2 props passed it is better to switch to new line separation
```suggestion
    <Pagination
      variant="minimal"
      currentPage={pageIndex + 1}
      pageCount={pageCount}
      paginationLabel="1234"
      onPageSelect={(pageNum) => gotoPage(pageNum - 1)}
    />
```

Comment (vlasovmichael) [reply src/DataTable/TablePaginationButtonGroup.jsx]: Fixed, thanks

File: src/DataTable/TablePaginationMinimal.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  const pageIndex = state?.pageIndex;
+
+  return (
+    <Pagination
+      variant="minimal"
+      currentPage={pageIndex + 1}
+      pageCount={pageCount}
+      paginationLabel="datatable pagination"
```
Comment (adamstankiewicz) [inline src/DataTable/TablePaginationMinimal.jsx:22]: nit: just "table pagination" should be fine here; don't think users need to know its a DataTable specifically.

File: src/Pagination/Pagination.scss [hunk 3]
```diff
@@ -1,15 +1,214 @@
...
+    button.page-link {
+      @extend .pgn__dark-styles;
+
+      &:focus {
+        box-shadow: none;
+      }
+    }
+
+    &:not(.active):focus {
+      box-shadow: 0 0 0 .0625rem $pagination-color;
```
Comment (adamstankiewicz) [inline src/Pagination/Pagination.scss:119]: Would it be possible to use an existing appropriate elevation SCSS variable here, e.g. `$level-1-box-shadow`?

File: src/Pagination/index.jsx [hunk 4]
```diff
@@ -278,23 +306,53 @@ class Pagination extends React.Component {
...
+    return (
+      <ul className="pagination">
+        {this.renderPreviousButton()}
+        {this.renderNextButton()}
+      </ul>
+    );
+  }
+
   render() {
+    const { variant, colorCheme, size } = this.props;
```
[the remark this answers is not in the document]
Comment (vlasovmichael) [reply src/Pagination/index.jsx]: changed to `invertColors `

File: src/Pagination/index.jsx [hunk 5]
```diff
@@ -369,12 +427,15 @@ Pagination.propTypes = {
...
+  colorCheme: PropTypes.oneOf(['light', 'dark']),
+  size: PropTypes.oneOf(['default', 'small']),
 };
 
 Pagination.defaultProps = {
   icons: {
-    leftIcon: <Icon className="fa fa-chevron-left mr-2" />,
-    rightIcon: <Icon className="fa fa-chevron-right ml-2" />,
+    leftIcon: <Icon className="fa fa-angle-left" />,
+    rightIcon: <Icon className="fa fa-angle-right" />,
```
[the remark this answers is not in the document]
Comment (vlasovmichael) [reply src/Pagination/index.jsx]: fixed. and I added advanced icons for bigger size

File: src/Pagination/index.jsx [hunk 6]
```diff
@@ -224,19 +252,19 @@ class Pagination extends React.Component {
           },
         )}
       >
-        <Button.Deprecated
+        <Button
```
[the remark this answers is not in the document]
Comment (vlasovmichael) [reply src/Pagination/index.jsx]: done

File: src/Pagination/index.jsx [hunk 7]
```diff
@@ -386,6 +447,17 @@ Pagination.defaultProps = {
...
+  size: 'default',
 };
 
+ReducedPagination.propTypes = {
+  currentPage: PropTypes.number.isRequired,
+  pageCount: PropTypes.number.isRequired,
+  handlePageSelect: PropTypes.func.isRequired,
+};
+
+Pagination.ReducedPagination = ReducedPagination;
```
[the remark this answers is not in the document]
Comment (vlasovmichael) [reply src/Pagination/index.jsx]: changed to `Pagination.Reduced`

File: src/DataTable/TablePaginationMinimal.jsx (hunk 2 above)
Comment (vlasovmichael) [reply src/DataTable/TablePaginationMinimal.jsx]: fixed

File: src/Pagination/Pagination.scss (hunk 3 above)
Comment (vlasovmichael) [reply src/Pagination/Pagination.scss]: fixed

File: src/Pagination/index.jsx [hunk 8]
```diff
@@ -184,34 +224,64 @@ class Pagination extends React.Component {
...
+                disabled={isFirstPage}
+              >
+                <div>
+                  {iconSize}
+                  {variant === VARIANTS.default ? buttonLabels.previous : null}
+                </div>
+              </Button>
+            )
+            : (
+              <IconButton
```
[the remark this answers is not in the document]
Comment (vlasovmichael) [reply src/Pagination/index.jsx:245]: @adamstankiewicz fixed. now all variants except default are with `IconButton`. please review

**Pull request merged**